server/src/main/java/io/spine/server/command/model/CommandHandlingClass.java

```diff
@@ -21,21 +21,31 @@
 package io.spine.server.command.model;
 
 import io.spine.core.CommandClass;
+import io.spine.type.MessageClass;
 
 import java.util.Set;
 
 /**
  * A common interface for classes that handle commands.
  *
- * @author Alexander Yevsyukov
+ * @param <P>
+ *         the type of message classes produced from the command handle
+ * @param <H>
+ *         the type of methods which perform command handling
  */
-public interface CommandHandlingClass<H extends CommandAcceptingMethod> {
+public interface CommandHandlingClass<P extends MessageClass<?>,
+                                      H extends CommandAcceptingMethod<?, P, ?>> {
 
     /**
      * Obtains classes of commands handled by the class.
      */
     Set<CommandClass> getCommands();
 
+    /**
+     * Obtains classes of all messages produced as a result of command handle.
+     */
+    Set<P> getHandleProducts();
```

Comment:
I'd like to avoid a `product` term. Let's rename this (and similar) occurrences to `getCommandOutput()` (similar case would be `getReactionOutput()` etc).